src/providers/vod/vod-provider.ts

```diff
@@ -12,20 +13,36 @@ export class VodProvider extends Provider {
   }
 
   _fetchVodData() {
-    let subTypesFilter = '';
+    let thumbSubTypesFilter = '';
     if (this._types.has(KalturaCuePointType.SLIDE)) {
-      subTypesFilter = `${KalturaThumbCuePointSubType.SLIDE},`;
+      thumbSubTypesFilter = `${KalturaThumbCuePointSubType.SLIDE},`;
     }
 
     // preparation for chapters
     // if (this._types.has(KalturaCuePoints.KalturaCuePointType.CHAPTER)) {
     //   subTypesFilter = `${subTypesFilter}${KalturaCuePoints.KalturaThumbCuePointSubType.CHAPTER},`;
     // }
 
-    const ks = this._player.config.session.ks;
-    const serviceUrl = this._player.config.provider.env.serviceUrl;
+    let requests: Array<ProviderRequest> = [];
+    if (thumbSubTypesFilter) {
+      requests.push({loader: ThumbLoader, params: {entryId: this._player.getMediaInfo().entryId, subTypesFilter: thumbSubTypesFilter}});
+    }
+    if (requests.length) {
+      this._player.provider
```

Comment:
move to     `if (thumbSubTypesFilter)` block and remove     `if (requests.length) `

Reply:
it is for future requests